src/models/graph.ts

```diff
@@ -263,18 +272,41 @@ export class Graph<N extends INodeBase, E extends IEdgeBase> implements IGraph<N
     this._settings?.onMergeData?.(data);
   }
 
-  // TODO(dlozic): Add delete all mechanic.
-  remove(data: Partial<{ nodeIds: number[]; edgeIds: number[] }>) {
+  remove(data: Partial<IGraphObjectsIds>) {
     const nodeIds = data.nodeIds ?? [];
     const edgeIds = data.edgeIds ?? [];
 
-    this._removeNodes(nodeIds);
-    this._removeEdges(edgeIds);
+    const removedNodesData = this._removeNodes(nodeIds);
+    const removedEdgesData = this._removeEdges(edgeIds);
+
+    const removedData = {
+      nodeIds: dedupArrays(removedNodesData.nodeIds, removedEdgesData.nodeIds),
+      edgeIds: dedupArrays(removedNodesData.edgeIds, removedEdgesData.edgeIds),
+    };
```

Comment:
Just a small detail here. Whenever you create a new object or an array, always add a type. Types will help you to raise warning if you for example change type in the future.

```
const removedData: IGraphObjectsIds = {
...
```

Also, this object should be within `if` on line 290 because you want to create this object only if settings exists and `settings.onRemoveData` exists, otherwise time and memory is wasted if settings is not called at all.